File: 백준/Silver/1931. 회의실 배정/회의실 배정.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Comparator;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        // 회의 수 입력
        int n = Integer.parseInt(br.readLine());
        int[][] meetings = new int[n][2];

        // 회의 시작 시간과 종료 시간 입력
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            meetings[i][0] = Integer.parseInt(st.nextToken()); // 시작 시간
            meetings[i][1] = Integer.parseInt(st.nextToken()); // 종료 시간
        }

        // 종료 시간 기준으로 정렬 (끝나는 시간, 시작 시간 순)
        Arrays.sort(meetings, new Comparator<int[]>() {
            @Override
            public int compare(int[] a, int[] b) {
                if (a[1] == b[1]) {
                    return Integer.compare(a[0], b[0]);
                }
                return Integer.compare(a[1], b[1]);
            }
        });

        // 회의 수 세기
        int count = 0;
        int lastEndTime = 0;

        // 회의 선택
        for (int i = 0; i < n; i++) {
            if (meetings[i][0] >= lastEndTime) { // 현재 회의의 시작 시간이 마지막 선택한 회의의 종료 시간과 같거나 클 경우
                count++;
                lastEndTime = meetings[i][1]; // 마지막 선택한 회의의 종료 시간을 업데이트
            }
        }

        // 결과 출력
        System.out.println(count);
    }
}

File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
# 회의실 배정 문제 풀이

# 회의 수 입력
n = int(input())

# 회의 정보를 저장할 리스트
meetings = []

# 회의 시작 시간과 종료 시간 입력
for _ in range(n):
    start, end = map(int, input().split())
    meetings.append((start, end))

# 종료 시간 기준으로 정렬 (끝나는 시간, 시작 시간 순)
meetings.sort(key=lambda x: (x[1], x[0]))

# 회의 수 세기
count = 0
last_end_time = 0

# 회의 선택
for start, end in meetings:
    if start >= last_end_time:  # 현재 회의의 시작 시간이 마지막 선택한 회의의 종료 시간과 같거나 클 경우
        count += 1
        last_end_time = end  # 마지막 선택한 회의의 종료 시간을 업데이트

# 결과 출력
print(count)
